app/models/oauth/token_exchange_grant.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module OAuth
+  ##
+  # Models token exchange grant.
+  class TokenExchangeGrant < ApplicationRecord
+    include OAuth::SessionCreatable
+
+    belongs_to :user
+    belongs_to :oauth_client, class_name: 'OAuth::Client'
+    has_many :oauth_sessions,
```

Comment:
❓ Are you planning on adding the `grant` polymorphic column to the `oauth_sessions` table in this PR? If not, you'll want to take this out.